neps/nep-0366.md

```diff
@@ -0,0 +1,148 @@
+---
+NEP: 366
+Title: Meta Transactions
+Author: Illia Polosukhin <ilblackdragon@gmail.com>
+DiscussionsTo: https://github.com/nearprotocol/neps/pull/366
+Status: Draft
+Type: Protocol Track
+Category: Runtime
+Created: 21-Jun-2022
+---
+
+## Summary
+
+In-protocol meta transactions allowing for third-party account to initiate and pay transaction fees on behalf of the account.
+
+## Motivation
+
+NEAR has been designed with simplicity of onboarding in mind. One of the large hurdles right now is that after creating an implicit or even named account the user does not have NEAR to pay gas fees to interact with apps.
+
+For example, apps that pay user for doing work (like NEARCrowd or Sweatcoin) or free-to-play games.
+
+[Aurora Plus](https://aurora.plus) has shown viability of the relayers that can offer some number of free transactions and a subscription model. Shifting the complexity of dealing with fees to the infrastructure from the user space.
+
+## Rationale and alternatives
+
+Proposed here design provides the easiest for users and developers way to onboard and pay for user transactions.
+There is no requirement on the user account, including user account may not even exist on chain and implicit account can be used.
+
+An alternative is proxy contracts deployed on the user account.
+This design has severe limitations as it requires user to deploy such contract and additional costs for storage.
+
+## Specification
+
+The main flow of the meta transaction will be as follows:
+ - User will sign `DelegateActionMessage` specifying set of actions that they need to be executed. It also includes specific relayer to ensure secure execution.
```

Comment:
```suggestion
 - User signs a `DelegateActionMessage` specifying the set of actions that they need to be executed. It will also specify a relayer to ensure secure execution.
```